File: leetcode-js/247-strobogrammatic-number-ii.js
/**
 * @param {number} n
 * @return {string[]}
 */
var findStrobogrammatic = function(n) {
    return helper(n, n);
};

function helper(n, target) {
    if (n === 0) {
        return [''];
    }
    if (n === 1) {
        return ['1', '8', '0'];
    }

    const list = helper(n - 2, target);
    const result = [];

    for (const str of list) {
        if (n !== target) {
            result.push(`0${str}0`);
        }
        result.push(`1${str}1`);
        result.push(`6${str}9`);
        result.push(`9${str}6`);
        result.push(`8${str}8`);
    }

    return result;
}

File: leetcode-java/247-strobogrammatic-number-ii.java
public class Solution {
    public List<String> findStrobogrammatic(int n) {
        return helper(n, n);
    }

    List<String> helper(int n, int target) {
        if (n == 0) {
            return Arrays.asList("");
        }

        if (n == 1) {
            return Arrays.asList("1", "8", "0");
        }

        List<String> list = helper(n - 2, target);
        List<String> res = new ArrayList<>();

        for (String num : list) {
            if (n != target) {
                res.add("0" + num + "0");
            }
            res.add("1" + num + "1");
            res.add("8" + num + "8");
            res.add("6" + num + "9");
            res.add("9" + num + "6");
        }

        return res;
    }
}
